Bottom: b42ef5c4092963b5c54c22b5153685c3fa0eb9aa
Top:    9c55d8dde24d153e54f4f0b96ab881ca76ed4fad
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-04 16:34:54 -0500

New role gitlab-user


---
diff --git a/roles/gitlab-user/README.md b/roles/gitlab-user/README.md
new file mode 100644
index 00000000..225dd44b
--- /dev/null
+++ b/roles/gitlab-user/README.md
@@ -0,0 +1,38 @@
+Role Name
+=========
+
+A brief description of the role goes here.
+
+Requirements
+------------
+
+Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance, if the role uses the EC2 module, it may be a good idea to mention in this section that the boto package is required.
+
+Role Variables
+--------------
+
+A description of the settable variables for this role should go here, including any variables that are in defaults/main.yml, vars/main.yml, and any variables that can/should be set via parameters to the role. Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
+
+Dependencies
+------------
+
+A list of other roles hosted on Galaxy should go here, plus any details in regards to parameters that may need to be set for other roles, or variables that are used from other roles.
+
+Example Playbook
+----------------
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+
+    - hosts: servers
+      roles:
+         - { role: username.rolename, x: 42 }
+
+License
+-------
+
+BSD
+
+Author Information
+------------------
+
+An optional section for the role authors to include contact information, or a website (HTML is not allowed).
diff --git a/roles/gitlab-user/defaults/main.yml b/roles/gitlab-user/defaults/main.yml
new file mode 100644
index 00000000..fd1bdcdd
--- /dev/null
+++ b/roles/gitlab-user/defaults/main.yml
@@ -0,0 +1,5 @@
+---
+# defaults file for gitlab-user
+
+gitlab_hostname: "127.0.0.1"
+gitlab_uri: "https://{{ gitlab_hostname }}/"
diff --git a/roles/gitlab-user/handlers/main.yml b/roles/gitlab-user/handlers/main.yml
new file mode 100644
index 00000000..1a06e329
--- /dev/null
+++ b/roles/gitlab-user/handlers/main.yml
@@ -0,0 +1,2 @@
+---
+# handlers file for gitlab-user
\ No newline at end of file
diff --git a/roles/gitlab-user/meta/main.yml b/roles/gitlab-user/meta/main.yml
new file mode 100644
index 00000000..72237997
--- /dev/null
+++ b/roles/gitlab-user/meta/main.yml
@@ -0,0 +1,57 @@
+galaxy_info:
+  author: your name
+  description: your description
+  company: your company (optional)
+
+  # If the issue tracker for your role is not on github, uncomment the
+  # next line and provide a value
+  # issue_tracker_url: http://example.com/issue/tracker
+
+  # Some suggested licenses:
+  # - BSD (default)
+  # - MIT
+  # - GPLv2
+  # - GPLv3
+  # - Apache
+  # - CC-BY
+  license: license (GPLv2, CC-BY, etc)
+
+  min_ansible_version: 1.2
+
+  # If this a Container Enabled role, provide the minimum Ansible Container version.
+  # min_ansible_container_version:
+
+  # Optionally specify the branch Galaxy will use when accessing the GitHub
+  # repo for this role. During role install, if no tags are available,
+  # Galaxy will use this branch. During import Galaxy will access files on
+  # this branch. If Travis integration is configured, only notifications for this
+  # branch will be accepted. Otherwise, in all cases, the repo's default branch
+  # (usually master) will be used.
+  #github_branch:
+
+  #
+  # platforms is a list of platforms, and each platform has a name and a list of versions.
+  #
+  # platforms:
+  # - name: Fedora
+  #   versions:
+  #   - all
+  #   - 25
+  # - name: SomePlatform
+  #   versions:
+  #   - all
+  #   - 1.0
+  #   - 7
+  #   - 99.99
+
+  galaxy_tags: []
+    # List tags for your role here, one per line. A tag is a keyword that describes
+    # and categorizes the role. Users find roles by searching for tags. Be sure to
+    # remove the '[]' above, if you add tags to this list.
+    #
+    # NOTE: A tag is limited to a single word comprised of alphanumeric characters.
+    #       Maximum 20 tags per role.
+
+dependencies: []
+  # List your role dependencies here, one per line. Be sure to remove the '[]' above,
+  # if you add dependencies to this list.
\ No newline at end of file
diff --git a/roles/gitlab-user/tasks/main.yml b/roles/gitlab-user/tasks/main.yml
new file mode 100644
index 00000000..5b538c01
--- /dev/null
+++ b/roles/gitlab-user/tasks/main.yml
@@ -0,0 +1,57 @@
+---
+# tasks file for gitlab-user
+
+- assert:
+    that:
+      - new_user is defined
+      - new_user_name is defined
+      - new_user_email is defined
+      - new_user_pass is defined
+
+# Login as root user
+- name: Get access token for admin user
+  uri:
+    url: "{{ gitlab_uri }}/oauth/token"
+    method: POST
+    body_format: json
+    body:
+      grant_type: password
+      username: root
+      password: "{{ gitlab_root_password }}"
+    return_content: yes
+  register: auth
+
+# Search for buildbot user
+- name: Find user
+  uri:
+    url: "{{ gitlab_uri }}/api/v4/users?username={{ new_user }}"
+    method: GET
+    headers:
+      Authorization: "Bearer {{ auth.json.access_token }}"
+    return_content: yes
+  register: founduser
+
+- name: Save new user
+  set_fact:
+    user: '{{ founduser.json[0] }}'
+  when: founduser.json|length > 0
+
+- name: Create missing user
+  block:
+  - uri:
+      url: "{{ gitlab_uri }}/api/v4/users"
+      method: POST
+      headers:
+        Authorization: "Bearer {{ auth.json.access_token }}"
+      return_content: yes
+      body_format: json
+      body:
+        username: "{{ new_user }}"
+        email: "{{ new_user_email }}"
+        password: "{{ new_user_pass }}"
+        name: "{{ new_user_name }}"
+    register: newuser
+
+  - set_fact:
+      user: '{{ newuser.json[0] }}'
+  when: founduser.json|length == 0
diff --git a/roles/gitlab-user/tests/inventory b/roles/gitlab-user/tests/inventory
new file mode 100644
index 00000000..878877b0
--- /dev/null
+++ b/roles/gitlab-user/tests/inventory
@@ -0,0 +1,2 @@
+localhost
+
diff --git a/roles/gitlab-user/tests/test.yml b/roles/gitlab-user/tests/test.yml
new file mode 100644
index 00000000..9a9a11c5
--- /dev/null
+++ b/roles/gitlab-user/tests/test.yml
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - gitlab-user
\ No newline at end of file
diff --git a/roles/gitlab-user/vars/main.yml b/roles/gitlab-user/vars/main.yml
new file mode 100644
index 00000000..612a06bd
--- /dev/null
+++ b/roles/gitlab-user/vars/main.yml
@@ -0,0 +1,2 @@
+---
+# vars file for gitlab-user
\ No newline at end of file
